obs2ioda-v2/src/f_c_string_i_mod.f90

```diff
@@ -0,0 +1,14 @@
+module f_c_string_i_mod
+    use iso_c_binding, only: c_ptr, c_size_t
+    implicit none
+
+    public :: strlen
+
+    interface
```

Comment:
Aside from unit tests, it looks like the only place this interface is used is in the `f_c_string_t_mod` module. It could reduce the number of source files and make the `f_c_string_t_mod` module self-contained to simply declare the `strlen` interface directly in the `f_c_string_t_mod` module.